android/app/src/main/AndroidManifest.xml

```diff
@@ -894,18 +894,30 @@
       </meta-data>
     </provider>
 
+    <service
+        android:name="app.organicmaps.car.NavigationCarAppService"
+        android:exported="true"
+        android:foregroundServiceType="location"
+        android:enabled="${ANDROID_AUTO_ENABLED}">
+      <intent-filter>
+        <action android:name="androidx.car.app.CarAppService" />
+        <category android:name="androidx.car.app.category.NAVIGATION" />
+      </intent-filter>
+    </service>
+
     <!-- Disable Google's checks for visited sites and loaded URLs in bookmarks description. -->
     <meta-data android:name="android.webkit.WebView.EnableSafeBrowsing" android:value="false" />
     <!-- Disable Google's anonymous stats collection -->
     <meta-data android:name="android.webkit.WebView.MetricsOptOut" android:value="true" />
 
-    <!-- For android Auto -->
-    <meta-data android:name="com.google.android.gms.car.application"
-        android:resource="@xml/automotive_app_desc"/>
+    <meta-data
+        android:name="com.google.android.gms.car.application"
+        android:resource="@xml/automotive_app_desc"
+        android:enabled="${ANDROID_AUTO_ENABLED}"/>
 
     <meta-data
         android:name="androidx.car.app.minCarApiLevel"
-        android:value="1"/>
-
+        android:value="1"
+        android:enabled="${ANDROID_AUTO_ENABLED}"/>
```

Comment:
Will AA code be compiled and embedded into the binary even if disabled?

Reply:
No, I disabled it (see above):

```
 if (!androidAutoEnabled) {
      sourceSets.main.java.excludes += '**/organicmaps/car'
    }
```

Reply:
> What (if any) influence of these changes to the production apk?

No, it is not enabled by default in the production APK:

```
./gradlew -Parm64 assembleGoogleRelease

> Configure project :app
Building with Google Mobile Services
Building without Google Firebase Services
Building without Android Auto
```

The feature flag works well:

./gradlew -Parm64 assembleGoogleBeta -Pandroidauto=true

103322785 Sep 30 10:31 ./app/build/outputs/apk/google/beta/OrganicMaps-23093003-google-beta.apk

./gradlew -Parm64 assembleGoogleBeta -Pandroidauto=false

102466536 Sep 30 10:35 ./app/build/outputs/apk/google/beta/OrganicMaps-23093004-google-beta.apk